pandas/tests/indexes/test_interval.py

```diff
@@ -491,6 +594,188 @@ def testcontains(self):
+        assert     Interval(1, 3, closed='both').overlaps(Interval(3, 5, closed='both'))
+
+        idx = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)])
+
+        assert     Interval(1, 3, closed='right').overlaps(idx) == np.array([1, 2])
+        assert     Interval(1, 2, closed='right').overlaps(idx) == np.array([2])
+        assert     Interval(0, 2, closed='right').overlaps(idx) == np.array([0, 2])
+        assert     Interval(3, 4, closed='right').overlaps(idx) == np.array([])
+
+        assert     Interval(1, 3, closed='left').overlaps(idx) == np.array([0, 1, 2])
+        assert     Interval(1, 2, closed='left').overlaps(idx) == np.array([0, 2])
+        assert     Interval(0, 2, closed='left').overlaps(idx) == np.array([0, 2])
+        assert     Interval(3, 4, closed='left').overlaps(idx) == np.array([3])
+
+        assert     Interval(1, 3, closed='both').overlaps(idx) == np.array([0, 1, 2])
+        assert     Interval(1, 2, closed='both').overlaps(idx) == np.array([0, 2])
+        assert     Interval(0, 2, closed='both').overlaps(idx) == np.array([0, 2])
+        assert     Interval(3, 4, closed='both').overlaps(idx) == np.array([3])
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_intervalIndex_covers(self):
+
+        # class IntervalIndex:
+        #     def covers(self, other: Interval) -> IntegerArray1D
+        #     def covers(self, other: IntervalIndex) -> Tuple[IntegerArray1D, IntegerArray1D]
+
+        idx = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)])
+
+        assert     idx.covers(Interval(1, 3, closed='right')) == np.array([1, 2])
+        assert     idx.covers(Interval(0, 3, closed='right')) == np.array([0, 1, 2])
+        assert     idx.covers(Interval(0, 2, closed='right')) == np.array([0])
+        assert     idx.covers(Interval(2, 4, closed='right')) == np.array([1])
+
+        assert     idx.covers(Interval(1, 3, closed='left')) == np.array([])
+        assert     idx.covers(Interval(0, 3, closed='left')) == np.array([0])
+        assert     idx.covers(Interval(0, 2, closed='left')) == np.array([0])
+        assert     idx.covers(Interval(2, 4, closed='left')) == np.array([1])
+
+        assert     idx.covers(Interval(1, 3, closed='both')) == np.array([1, 2])
+        assert     idx.covers(Interval(0, 5, closed='both')) == np.array([0, 1, 2])
+        assert     idx.covers(Interval(0, 2, closed='both')) == np.array([0])
+        assert     idx.covers(Interval(2, 4, closed='both')) == np.array([1])
+
+        idx1 = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='right')
+        idx2 = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='left')
+        idx3 = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='both')
+
+        assert     idx.covers(idx1) == (np.array([0,1,2,2]), np.array([0,1,1,2]))  # note: I had to choose an arbitrary ordering. If this test fails, double check the test too...
```

Comment:
You mean the order of the last two entries here? I'm actually OK with this -- let's say that returning indices in lexicographically-sorted order is part of the spec.